build/ci.go

```diff
@@ -158,6 +158,13 @@ func doInstall(cmdline []string) {
 		build.MustRun(goinstall)
 		return
 	}
+	// If we are cross compiling to ARMv5 ARMv6 or ARMv7, clean any prvious builds
+	if *arch == "arm" {
+		os.RemoveAll(filepath.Join(runtime.GOROOT(), "pkg", runtime.GOOS+"_arm"))
+		for _, path := range filepath.SplitList(build.GOPATH()) {
+			os.RemoveAll(filepath.Join(path, "pkg", runtime.GOOS+"_arm"))
+		}
+	}
```

Comment:
Suggestion: maybe we should add a cross build subcommand that uses `go build` instead of `go install`. That would remove the weird GOBIN avoidance logic and would also eliminate this hack.